src/lib/accounts.js

```diff
@@ -38,3 +38,18 @@ export function getAccountsByType (cozy, accountType, accountsIndex) {
     })
   }
 }
+
+export function getAllAccounts (cozy, accountsIndex) {
+  if (accountsIndex) {
```

Comment:
Maybe something like that helps to have simpler code ?

```js
const getIndex  = accountsIndex ? Promise.resolve(accountsIndex) : indexAccountsByType(cozy)
getIndex.then(index => {
     return cozy.data.query(index, {
         selector: {'account_type': {'$gt': null}}
       })
    })
```

This way you don't have to repeat the query part if it changes.